purchase_stock_picking_return_invoicing/models/purchase_order.py

```diff
@@ -68,37 +68,60 @@ def action_view_invoice_refund(self):
         action = self.env.ref("account.action_move_in_refund_type")
         result = action.read()[0]
         create_refund = self.env.context.get("create_refund", False)
+        # Invoice_ids may be filtered depending on the user. To ensure we get all
+        # invoices related to the purchase order, we read them in sudo to fill the
+        # cache.
+        self.sudo()._read(["invoice_ids"])
```

Comment:
I don't get the need for this

Reply:
Because the core method `action_view_invoice` also has it.

Reply:
Then put that and an absolute link to the reference in Odoo core.

Reply:
https://github.com/odoo/odoo/blob/13.0/addons/purchase/models/purchase.py#L448

Reply:
I mean to put it in the code:
```suggestion
        # Mimic upstream code
        # https://github.com/odoo/odoo/blob/3572ce3a06bad/addons/purchase/models/purchase.py#L448
        self.sudo()._read(["invoice_ids"])
```

Reply:
ah, ok